bld/src/libproxy/CMakeLists.txt

```diff
@@ -84,55 +86,75 @@ else()
     if(HAVE_LINUX_RANDOM_H)
         set(_PAL_COMPONENTS ${_PAL_COMPONENTS} pal_rand_linux)
         message(STATUS "  linking with urandom rand")
-   else()
+    else()
         set(use_openssl 1)
         set(_PAL_COMPONENTS ${_PAL_COMPONENTS} pal_rand_openssl)
         message(STATUS "  linking with openssl rand")
     endif()
 
-    pkg_check_modules(LIBSECRET libsecret-1)
-    if(LIBSECRET_FOUND)
-        set(_PAL_COMPONENTS ${_PAL_COMPONENTS} pal_cred_gnome)
-        set(_PAL_DEP_LIBS ${_PAL_DEP_LIBS} ${LIBSECRET_LIBRARIES})
-        set(_PAL_DEP_INC ${_PAL_DEP_INC} ${LIBSECRET_INCLUDE_DIRS})
-        set(_PAL_DEP_DEFS ${_PAL_DEP_DEFS} __HAVE_LIBSECRET=1)
-        message(STATUS "  linking with libsecret-1")
+    if(CMAKE_CROSSCOMPILING)
+        # As mentioned at https://cmake.org/Wiki/CMake_Cross_Compiling the
+        # pkg-config tool can not be used by cmake while cross compiling.
```

Comment:
Cross compiling fails, because the following line is missing:

set(_PAL_COMPONENTS ${_PAL_COMPONENTS} pal_cred_stub)

So the linking fails with:
Linking C executable ../../../bin/proxyd
../../../lib/libproxy.so: undefined reference to `pal_cred_hmac_sha256'
../../../lib/libproxy.so: undefined reference to `pal_cred_deinit'
../../../lib/libproxy.so: undefined reference to `pal_cred_remove'
../../../lib/libproxy.so: undefined reference to `pal_cred_init'
../../../lib/libproxy.so: undefined reference to `pal_cred_import'